### PR TITLE
DEVEX-293: Verification : use dedicated function for 'report' and 'status'

### PR DESCRIPTION
**Pull request by JPPortier**

### CODE REVIEW

File: client/src/test/java/com/sinch/sdk/domains/verification/adapters/VerificationsServiceTest.java [hunk 1]
```diff
@@ -51,28 +66,104 @@ public void initMocks() {
...
+        .usingRecursiveComparison()
+        .isEqualTo(VerificationsDtoConverterTest.expectedStartVerificationSeamlessResponse);
+  }
+
+  @Test
+  void reportSmsByIdentity() throws ApiException {
+
+    when(api.reportVerificationByIdentity(
+            eq("number"), eq("endpoint string"), eq(verificationReportSMSRequestDto)))
+        .thenReturn(VerificationReportDtoTest.expectedVerificationCalloutDto);
```
Comment (asein-sinch) [inline client/src/test/java/com/sinch/sdk/domains/verification/adapters/VerificationsServiceTest.java]: Should be a SMS response

File: client/src/test/java/com/sinch/sdk/domains/verification/adapters/VerificationsServiceTest.java [hunk 2]
```diff
@@ -51,28 +66,104 @@ public void initMocks() {
...
+
+    VerificationReport response =
+        service.reportSmsByIdentity(
+            NumberIdentity.builder().setEndpoint("endpoint string").build(),
+            VerificationsDtoConverterTest.verificationReportSMSRequest);
+
+    Assertions.assertThat(response)
+        .usingRecursiveComparison()
+        .isEqualTo(VerificationsDtoConverterTest.expectedVerificationReportCalloutResponse);
```
Comment (asein-sinch) [inline client/src/test/java/com/sinch/sdk/domains/verification/adapters/VerificationsServiceTest.java]: Should be the SMS response returned above on line 129
(check other methods for reportById and reportByIdentity which all use a callout)

**Review by Dovchik — DISMISSED** [review]
LGTM

File: client/src/test/java/com/sinch/sdk/domains/verification/adapters/VerificationsServiceTest.java (hunk 1 above)
Comment (JPPortier) [reply client/src/test/java/com/sinch/sdk/domains/verification/adapters/VerificationsServiceTest.java]: Of course !
Thank you
Fixed(and added missing proper dedicated type returned by service APIs)

File: client/src/test/java/com/sinch/sdk/domains/verification/adapters/VerificationsServiceTest.java (hunk 2 above)
Comment (JPPortier) [reply client/src/test/java/com/sinch/sdk/domains/verification/adapters/VerificationsServiceTest.java]: Fixed

**Review by asein-sinch — APPROVED** [review]

**Pull request merged**